core/src/databases/remote_databases/snowflake.rs

```diff
@@ -117,8 +119,9 @@ impl TryFrom<SnowflakeRow> for QueryResult {
 }
 
 impl SnowflakeRemoteDatabase {
-    pub fn new(secret: &str) -> Result<Self> {
-        let connection_details: SnowflakeConnectionDetails = serde_json::from_str(secret)?;
+    pub fn new(credentials: serde_json::Map<String, serde_json::Value>) -> Result<Self> {
+        let connection_details: SnowflakeConnectionDetails =
+            serde_json::from_value(serde_json::Value::Object(credentials))?;
```

Comment:
You can't directly use serde_json::from_value(credentials) ?

Reply:
no

Reply:
from_value needs a value. But wrapping the map in a value is free